cli/src/lib.rs

```diff
@@ -627,19 +636,22 @@ fn init(
     fs::write("Anchor.toml", toml)?;
 
     // Build virtual manifest.
-    fs::write("Cargo.toml", template::virtual_manifest())?;
+    fs::write("Cargo.toml", rust_template::virtual_manifest())?;
 
     // Initialize .gitignore file
-    fs::write(".gitignore", template::git_ignore())?;
+    fs::write(".gitignore", rust_template::git_ignore())?;
 
     // Initialize .prettierignore file
-    fs::write(".prettierignore", template::prettier_ignore())?;
+    fs::write(".prettierignore", rust_template::prettier_ignore())?;
 
     // Build the program.
+    fs::create_dir("solidity")?;
     fs::create_dir("programs")?;
```

Comment:
No, it does not. However, we want solidity programs and rust programs to co-exists in the anchor workspace. So, it makes sense to have it in case some rust programs get added.

Reply:
No, it does not. I've removed it. Now if you create an solidity workspace with `anchor init -s` and then add a rust program with `anchor new`, then `programs` gets created if does not yet exists (using `fs::create_dir_all()`)